src/rgw/rgw_d3n_cacherequest.h

```diff
@@ -45,12 +41,27 @@ struct D3nL1CacheRequest : public D3nCacheRequest {
   int stat;
   int ret;
   struct aiocb* paiocb;
+
   D3nL1CacheRequest() :  D3nCacheRequest(), stat(-1), paiocb(nullptr) {}
-  ~D3nL1CacheRequest(){}
+  ~D3nL1CacheRequest(){
+    lock.lock();
+    if (paiocb != nullptr) {
+      if (paiocb->aio_buf != nullptr) {
+        free((void*)paiocb->aio_buf);
+        paiocb->aio_buf = nullptr;
+      }
+      ::close(paiocb->aio_fildes);
+      delete(paiocb);
+    }
+    lock.unlock();
+
+    lsubdout(g_ceph_context, rgw_datacache, 30) << "D3nDataCache: " << __func__ << "(): Read From Cache, comlete" << dendl;
+  }
 
   int execute_io_op(std::string obj_key, bufferlist* bl, int read_len, int ofs, int read_ofs, std::string& cache_location,
-                 void(*f)(sigval_t), rgw::Aio* aio, rgw::AioResult* r) {
+                    void(*f)(sigval_t), rgw::Aio* aio, rgw::AioResult* r) {
```

Comment:
I would create a typedef or using synonym for the callback parameter, I think

Reply:
ack